lite2/client.go

```diff
@@ -1110,6 +1124,26 @@ func (c *Client) validatorSetFromPrimary(height int64) (*types.ValidatorSet, err
 	return c.validatorSetFromPrimary(height)
 }
 
+func (c *Client) signedHeaderFromWitness(witness provider.Provider, height int64) (*types.SignedHeader, error) {
+	for attempt := uint16(1); attempt <= c.maxRetryAttempts; attempt++ {
```

Comment:
can we extract this code into a function (smth like `signedHeaderFrom(p Provider, ...)`)?

Reply:
Yes, we can try. You mean to also merge it with the signedHeaderFromPrimary. I thought about earlier but just decided to implement it like this first

Reply:
> You mean to also merge it with the signedHeaderFromPrimary

yes

Reply:
Okay yes I can do that

Reply:
In hindsight, I don't think that these two should be merged because they are from distinct data structures and they have different processes surrounding how non responsiveness should be dealt with